conda_build/source.py

```diff
@@ -67,16 +70,16 @@ def download_to_cache(cache_folder, recipe_path, source_dict, verbose=False):
         source_dict["fn"] if "fn" in source_dict else basename(source_urls[0])
     )
     hash_added = False
-    for hash_type in ("md5", "sha1", "sha256"):
+
+    for hash_type in ACCEPTED_HASH_TYPES:
         if hash_type in source_dict:
```

Comment:
Let's only loop over the hash types present (this will also make the `else` clause work correctly again now that `break` has been removed):

```suggestion
    for hash_type in set(source_dict).intersection(ACCEPTED_HASH_TYPES):
```